server/storage/database.go

```diff
@@ -127,7 +127,23 @@ func (db *SQLStorage) UpdateMany(gun string, updates []MetaUpdate) error {
 func (db *SQLStorage) GetCurrent(gun, tufRole string) ([]byte, error) {
 	var row TUFFile
 	q := db.Select("data").Where(&TUFFile{Gun: gun, Role: tufRole}).Order("version desc").Limit(1).First(&row)
+	return returnRead(q, row)
+}
+
+// GetChecksum gets a specific TUF record by its hex checksum
+func (db *SQLStorage) GetChecksum(gun, tufRole, checksum string) ([]byte, error) {
```

Comment:
I don't believe so. It shouldn't be the job of a storage engine to know how long a checksum should be or which hashing methods are valid.